src/scripts/main.js

```diff
@@ -1,3 +1,42 @@
 'use strict';
 
-// write code here
+const addRow = document.querySelector('.append-row');
+const removeRow = document.querySelector('.remove-row');
+const addColumn = document.querySelector('.append-column');
+const removeColumn = document.querySelector('.remove-column');
+
+const bodyTable = document.querySelector('table');
+
+const allRows = bodyTable.getElementsByTagName('tr');
+
+addRow.addEventListener('click', (e) => {
+  if (allRows.length < 10) {
+    const newRow = allRows[allRows.length - 1].cloneNode(true);
```

Comment:
Cloning a row with `cloneNode(true)` will duplicate all child nodes, including any event listeners attached to them. If the cloned row contains interactive elements with event listeners, this could lead to unexpected behavior. Consider creating a new row and manually copying the necessary content or attributes.